clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -291,6 +303,45 @@ private void completeFutureAndFireCallbacks(long baseOffset, long logAppendTime,
         return batches;
     }
 
+    public boolean completed() {
```

Comment:
This function is only used in unit test, we should not expose.

Reply:
Done